packages/drivers/routerlicious-driver/src/deltaStorageService.ts

```diff
@@ -54,17 +56,17 @@ export class DeltaStorageService implements IDeltaStorageService {
         to?: number): Promise<ISequencedDocumentMessage[]> {
         const query = querystring.stringify({ from, to });
 
-        let headers: { Authorization: string } | null = null;
+        const headers: OutgoingHttpHeaders = {
+            "x-correlation-id": uuid.v4(),
```

Comment:
What's the value of this ID?
I can see the value if it was written to client side telemetry, then we can use it to correlate activity between client & server. But with this code, it will land only on server, so server can generate ID with the same result, no?

Reply:
That's a good point. We added correlation id in server side and this PR is just for the completeness for the server side to receive http requests with correlation id in the header. As for client side telemetry, I didn't go deep into that. Could you explain a little bit more how we can add to client side telemetry? Or maybe we can talk offline with Tanvir.

Reply:
@vladsud Correct! FRS has recently done the work to store and track correlation ids of http requests. Next, we want wrap all these calls with telemetry. Can you provide us some pointers on how ODSP driver does that? I am expecting to follow a similar pattern.

Reply:
Take a look at how ITelemetryLogger / ITelemetryBaseLogger and usage in ODSP driver.
Basically it is passed to driver in Container.load() (when calling createDocumentService) and driver passes it through it's layers and uses to log data (look for this.logger in odspDocumentStorageManager.ts as an example).
We wrap pretty much every network call and write out a bunch of properties to allow us to do analyzes later.

Note that in similar case, we rely on server communicating to client its correlation ID (in http response), and client recording it in client-side logging. That way, server is pretty much agnostic of client and client behavior. Reverse would work too, I do not have strong opinion here.